Src/ILGPU/Runtime/MemoryBuffers.tt

```diff
@@ -717,5 +721,53 @@ namespace ILGPU.Runtime
         #endregion
     }
 
+    public partial class MemoryBufferExtensions
+    {
+        /// <summary>
+        /// Allocates a <#= i #>D memory buffer with the given content on the associated
+        /// accelerator.
+        /// </summary>
+        /// <typeparam name="T">The element type.</typeparam>
+        /// <param name="accelerator">The current accelerator.</param>
+        /// <param name="data">The initial data array.</param>
+        /// <returns>The allocated memory buffer.</returns>
+<#      if (i > 1) { #>
+        [SuppressMessage(
+            "Microsoft.Performance",
+            "CA1814: PreferJaggedArraysOverMultidimensional",
+            Target = "data")]
+<#      } #>
+<#      var allocExpression = Enumerable.Range(0, i).Select(t => $"data.GetLength({i})"); #>
```

Comment:
```suggestion
<#      var allocExpression = Enumerable.Range(0, i).Select(t => $"data.GetLength({t})"); #>
```

ISSUE: This is incorrectly generating code like: `var buffer = accelerator.Allocate<T>(data.GetLength(3), data.GetLength(3), data.GetLength(3));`

Reply:
That's why code reviews are important 😄